### PR TITLE
Deprecate the preload argument to `find`/`fetchById` and move in into the `…

### PR DESCRIPTION
**Pull request by bmac**
…preload` key on the options argument.

Closes https://github.com/emberjs/data/issues/3248

### CODE REVIEW

File: packages/ember-data/lib/system/store.js [hunk 1]
```diff
@@ -584,21 +584,43 @@ Store = Service.extend({
...
+    if (options) {
+      var modelProperties = [];
+      var fields = Ember.get(internalModel.type, 'fields');
+      fields.forEach(function(fieldType, key) {
+        modelProperties.push(key);
+      });
+      var preloadDetected = modelProperties.reduce(function(memo, key) {
+        return typeof options[key] !== 'undefined' || memo;
+      }, false);
+      if (preloadDetected) {
```
Comment (igorT) [inline packages/ember-data/lib/system/store.js]: This should not be on `findRecord` as findRecord is a new api, we should add the deprecation only on `find` and `findById` if findById was public

Comment (bmac) [inline packages/ember-data/lib/system/store.js]: Updated.

**Comment by igorT** [comment]
Rebase please? :(

**Comment by bmac** [comment]
Rebased

**Comment by igorT** [comment]
Can you please add a test that confirms that the deprecated form still works?

**Pull request merged**